src/Symfony/Component/HttpKernel/Debug/FileLinkFormatter.php

```diff
@@ -22,11 +21,11 @@
 class FileLinkFormatter implements \Serializable
 {
     private $fileLinkFormat;
-    private $requestStack;
+    private $router;
     private $baseDir;
-    private $urlFormat;
+    private $queryString;
 
-    public function __construct($fileLinkFormat = null, RequestStack $requestStack = null, $baseDir = null, $urlFormat = null)
+    public function __construct($fileLinkFormat = null, UrlGeneratorInterface $router = null, $baseDir = null, $queryString = null)
```

Comment:
The type change is a BC break. I saw the comment saying we don't care because this is debug tool, but I don't think we've done any such BC break before (we did some for the profiler, but they were behavior ones, not "interface" ones AFAIK.)